### ISSUE TITLE
[Iteration enhancement] exception while updating iterated node name

### ISSUE THREAD

**Issue by LStruber** [populse/populse_mia#241]
An exception is not handled when trying to change iterated node name.

Minimum steps to reproduce:
- Create an iterated pipeline (on a smooth for example)
- Try to change the node name of the iterated node in the node controller

MIA close itself with an Exeption in the standard output:
```
Traceback (most recent call last):
  File "/home/lucas/DEV/populse_dev/populse_mia/python/populse_mia/user_interface/pipeline_manager/node_controller.py", line 562, in update_node_name
    old_node = pipeline.nodes[old_node_name]
KeyError: 'iterated_smooth_1'
```

Two solutions here:
- Do not allow the user to change an iterated pipeline name (catching the exception, graying out the name in the node controller for iterated node, or popup an error "impossible to edit iterated pipeline names")
- Allow the user to change its name, but we need to be careful here because in several places we check if the node is iterated by checking if its name contains "iterated_", so the new name must be prepend by "iterated_" if not done by the user

**Comment by servoz** [comment]
The 2 solutions are OK for me.

- Preventing the name of an iterated pipeline from being changed is acceptable, especially since currently iteration only works for a whole pipeline, so the name change may not seem very interesting.
-  Prepending by "iterated_" is a fairly simple solution (well it seems to me, sometimes what seems simple turns out to be much more complicated when we start looking at what it involves in  coding).

So I would tend to think that we should implement what will be the fastest to achieve!

**Comment by LStruber** [comment]
fix with commit 8c7d0327367d1312e7d6abb8e58d6e3866ee3b05 (solution 2).
